example/javascriptlib/linting/1-linting/build.mill

```diff
@@ -0,0 +1,56 @@
+package build
+
+import mill._, javascriptlib._
+
+object foo extends TypeScriptModule
+
+// Mill supports code linting via `eslint` https://eslint.org, `typescript-eslint` https://typescript-eslint.io/getting-started
+// and `prettier` https://prettier.io/docs/en out of the box.
+// You can lint your projects code by providing a configuration for your preferred linter and running `mill _.checkFormatAll`.
+
+// If both configuration files are present, the command `mill _.checkFormatAll` will default to eslint.
+// If neither files are present, the command will exit with an error, you must include at least one configuration file.
+// You can lint via a specificied linter via the commands `mill _.checkFormatEslint` to lint with eslint and
+// `mill _.checkFormatPrettier` to lint with prettier.
+
+// When using prettier you can specify the path to lint via command line argument, `mill _.checkFormatAll "*/**/*.ts"`
+// just as you would when running `prettier --check` if no path is provided mill will default to using "*/**/*.ts".
+// Also if a `.prettierignore` is not provided, mill will generate one ignoring "node_modules" and ".git".
+
+// You can define `npmLintDeps` field to add dependencies specific to linting to your module.
+// The task `npmLintDeps` works the same way as `npmDeps` or `npmDevDeps`.
+
+/** Usage
+> cat foo/src/foo.ts # initial poorly formatted source code
+export class Foo{
+static main(
+args: string[
+])
+{console.log("Hello World!")
+}
+}
+
+> mill foo.checkFormatAll # run linter - since both eslint and prettier configurations are present, mill will opt to use eslint.
```

Comment:
Let's move the default `checkFormatAll` command to an `ExternalModule`, similar to `ScalafmtModule`/`PalantirFormatModule`/`KtlintModule`. That will help standardize the user experience across Mill language toolchains